third_party/amd/lib/TritonAMDGPUToLLVM/SchedInstructions.cpp

```diff
@@ -0,0 +1,191 @@
+#include "TritonAMDGPUToLLVM/Passes.h"
+
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Pass/Pass.h"
+#include "third_party/amd/include/Dialect/TritonAMDGPU/IR/Dialect.h"
+#include "triton/Dialect/Triton/IR/Dialect.h"
+
+namespace mlir::triton {
+#define GEN_PASS_DEF_INSERTINSTRUCTIONSCHEDHINTS
+#define GEN_PASS_DEF_LOWERINSTRUCTIONSCHEDHINTS
+#include "TritonAMDGPUToLLVM/Passes.h.inc"
+} // namespace mlir::triton
+
+using namespace mlir;
+
+namespace {
+
+// The bitmask that encodes kinds of the instructions from AMD ISA.
+// The bitmask is used for providing instruction scheduling hints.
+enum InstructionKindMask {
```

Comment:
Prefer to use `enum class` to require explicit prefix--right now I belive `None` alone works..

Reply:
@antiagainst, Let's not use `enum class` here. `InstructionKindMask` is in the scope of the `SchedInstructions.cpp` translation unit and it is only for the internal use. This C-enum is convenient because we can use logical `AND`; thus we can conveniently combine multiple kinds of instructions in one `sched.group.barrier` intrinsic call. For example,

```cpp
      createSchedBarrier(rewriter, InstructionKindMask::DS_READ | InstructionKindMask::DS_WRITE);
```

Reply:
Sure. It's a minor thing I'm fine with it.